Add unit test for #121: AddressList method fluency

braindawg opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->
This PR provides a failing unit test to demonstrate Issue #121.

braindawg committed:
Add unit test for #121: AddressList method fluency

This adds a failing unit test to accompany issue #121. The
AddressList::addFromString method should be chainable, and is not.

Signed-off-by: Brian Dunne <braindawg@gmail.com>

Ocramius commented on test/AddressListTest.php:
```diff
@@ -116,6 +117,18 @@ public function testCanAddManyAddressesAtOnce()
+        try {
+            $this->list->addFromString('test_fromstring_fluency1@example.com')
+                ->addFromString('test_fromstring_fluency2@example.com');
+        } catch (Throwable $e) {
```
This `catch` is not necessary

braindawg replied:
Took the cue on that from a couple other unit tests in AddressListTest.php that were similarly `catching` their errors. Removed mine!

Ocramius reviewed (changes requested):
Let's target 2.12.x also here, since it's a bugfix

braindawg committed:
Remove unnecessary catch from unit test

Per ocramius's suggestion, removed a superfluous try/catch block in a
unit test.

Signed-off-by: Brian Dunne <braindawg@gmail.com>

Ocramius commented:
Thanks @braindawg!
